Userland/Libraries/LibCore/Stream.cpp

```diff
@@ -165,6 +165,31 @@ ErrorOr<size_t> File::read(Bytes buffer)
     return nread;
 }
 
+ErrorOr<ByteBuffer> File::read_all()
+{
+    off_t file_size = 0;
+    struct stat st;
+    int rc = fstat(m_fd, &st);
+    if (rc == 0)
+        file_size = st.st_size;
+
+    ByteBuffer data;
+    data.ensure_capacity(file_size);
+
+    size_t total_read {};
+    while (true) {
+        auto chunk = TRY(data.get_bytes_for_writing(4096));
+        auto nread = TRY(read(chunk));
+
+        total_read += nread;
+        data.resize(total_read);
```

Comment:
Why resize here? You can simply resize once after the while loop is exited.

Reply:
ByteBuffer::get_bytes_for_writing ensure capacity, but doesn't change the size.
So each call in the loop will return a Bytes object with the same properties, unless we call resize() each time

Reply:
Hm, that seems like a bug.

Reply:
Should be good now

Reply:
Yup, LGTM :+1: